lib/tuya.js

```diff
@@ -1387,6 +1399,82 @@ const valueConverter = {
+    heatOff: valueConverterBasic.lookup({'heat': true, 'off': false}),
+    heatIdle: valueConverterBasic.lookup({'heat': 1, 'idle': 0}),
+    lookupIgnoreError: (map) => {
+        return {
+            from: (v) => {
+                try {
+                    return valueConverterBasic.lookup(map).from(v);
+                } catch (e) {
+                    // Don't change the state if invalid data is received from the device
+                }
+            },
+            to: (v) => {
+                return valueConverterBasic.lookup(map).to(v);
+            },
+        };
+    },
+    weekFormat: valueConverterBasic.lookup({'5+2': new Enum(0), '6+1': new Enum(1), '7': new Enum(2)}),
+    scheduleHoursMinutes: (_idx) => {
+        return {
+            from: (v) => {
+                const hoursRaw = v[_idx];
+                const minutesRaw = v[_idx + 1];
+                const hours = hoursRaw.toString().padStart(2, '0');
+                const minutes = minutesRaw.toString().padStart(2, '0');
+                return [hours, minutes];
+            },
+            to: (v) => {
+                const [h, m] = v;
+                if (h < 0 || h >= 24) throw new Error('Invalid hour of: ' + h);
+                if (m < 0 || m >= 60) throw new Error('Invalid minute of: ' + m);
+                return v;
+            },
+        };
+    },
+    scheduleTemp: (_idx) => {
+        return {
+            from: (v) => {
+                const tempRaw = convertNByteHexArrayToDecimalValue([v[_idx + 2], v[_idx + 3]]);
+                return tempRaw / 10;
+            },
+            to: (v) => {
+                if (v < 5 || v > 60 || v % 0.5 !== 0) throw new Error('Invalid temperature of: ' + v);
+                return convertDecimalValueTo2ByteHexArray(v * 10);
+            },
+        };
+    },
+    // need to refactor
+    schedule: (periodsNumber, timeConverter, tempConverter) => {
+        return {
+            from: (v) => {
+                const schedule = [];
+
+                for (let i = 0; i < periodsNumber; i++) {
+                    const [hours, minutes] = timeConverter(i).from(v);
+                    const temp = tempConverter(i).from(v);
+                    schedule.push(`${hours}:${minutes}/${temp}`);
+                }
+
+                return schedule.join(' ');
+            },
+            to: (v) => {
+                const schedule = v.split(' ');
+                const payload = [];
+
+                for (let i = 0; i < periodsNumber; i++) {
+                    const [timeRaw, tempRaw] = schedule[i].split('/');
+                    const [hoursRaw, minutesRaw] = timeRaw.split(':', 2);
+                    const time = timeConverter(i).to([parseInt(hoursRaw), parseInt(minutesRaw)]);
+                    const temp = tempConverter(i).to(parseFloat(tempRaw));
+                    payload.push(...time, ...temp);
+                }
+
+                return payload;
+            },
+        };
+    },
```

Comment:
Here's an attempt to unify Tuya's schedule. The main idea is to create a single schedule format: hours1:minutes1/temp1 hours2:minutes2/temp2 ... and, if there is a necessity to custom process the time or temperature, converters can be created especially for this or used already defined.
What do you think?

Reply:
Instead of a string we could also expose it as a composite. e.g. https://github.com/Koenkk/zigbee-herdsman-converters/pull/4775/files#diff-2d16489d92b9b8bfd6140c0045ff1f2f7e9caf7df7e92445313f1d1a7609da1aR2598 (without the composite being nested in a list)